joshuaCarpentier/assignments/treesDFS/lc113/lc113.js

```diff
@@ -0,0 +1,82 @@
+// Problem Statement #
+
+// Given a binary tree and a number ‘S’, find all paths from root-to-leaf such
+// that the sum of all the node values of each path equals ‘S’.
+
+// Example
+// -----
+// targetSum = 12
+// Output = [ [ 1, 7, 4 ], [ 1, 9, 2 ] ]
+//      1
+//     / \
+//    7   9
+//  / |   | \
+// 4  5   2  7
+
+
+
+
+class TreeNode {
+  constructor(value) {
+    this.value = value;
+    this.left = null;
+    this.right = null;
+  }
+};
+
+
+/**
+ * 
+ * Input: tree
+ * Output: Boolean value weather the totalSum of a path equals the targetSum
+ *  
+ * Naive Approach
+ * --------------- 
+ * Variables: root, root. left, root.right, index, array to store the root essentially a stack, TotalSum
+ * 
+ * 
+ */
+// Test Cases are conditions for inputs 
+const find_paths = (root, sum) =>{
+  const search = (tree, parents = [], parentsSum = 0) => {
+    // Pre-Condition: check if exist
+    if (!tree) return false;
+    const currSum = parentsSum + tree.value;
+    const currArr = [...parents, tree.value]; // what does the three dots do
```

Comment:
The three dots make a copy of the parents arr and "spread the values" across your currArr. Here's a link: https://www.javascripttutorial.net/es6/javascript-spread/